lib-es5/utils/index.js

```diff
@@ -1116,6 +1116,20 @@ function download_zip_url() {
   }));
 }
 
+/**
+ * Returns a URL that when invokes creates an archive of a folder
+ */
+function download_folder(folder_path) {
+  var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
+
+  options.resource_type = "all";
+  options.prefixes = folder_path;
+  var cloudinary_params = exports.sign_request(exports.archive_params(merge(options, {
```

Comment:
How this works? the signature include the `mode='download'` but the options sent to the api does not

Reply:
its part of the cloudinary_params sent in the hashToQuery(cloudinary_params);

Its the same concept needed in download_archive_url- https://github.com/cloudinary/cloudinary_npm/blob/b1e23fc7c0905777c9c34bf328405b85a00b3577/lib/utils/index.js#L1009

Reply:
Still don't understand how this works

Reply:
What do you mean? both the signature and the api call include the `mode=download param`.

The `sign_request` method returns the signed params with the `mode` param included (after signing).

Reply:
What am I missing? 
There are `options` which are sent to the API and when building the signature, we merge `options` with `mode=download`

Reply:
After discussion - all params are sent on the query string and the options sent to the `api_url` are for general config and not params to the specific api call